### DISCUSSION TITLE
average filesize if pyinstaller is used to deploy application

### DISCUSSION THREAD

**Discussion by retsyo** [zauberzeug/nicegui#1160] [Q&A]
I have tried to use pyinstaller to bundle examples/trello_cards on windows 10 64bits. I get a ~700M monster which depends on numpy, matplotlib, so it is not a feasible filesize if I give my application to end users.
1. what is the average size if pyinstaller is used?
2. any way to make the file size smaller, 20~30M is good, 100M is acceptable, >100M is a mad result

Thanks

**Comment by CatamountJack** [comment]
I believe #1143 in the next 1.3 release will allow make things like matplotlib and numpy optional if we don't need them - which should significantly reduce the package size.  I just checked one of the programs I built and it's about 130 MB.  Still too big, but definitely less than 700 MB.  Make sure your app is running in a virtualenv with only those packages that you are actually using and that you're not accidentally importing anything in your program you don't use.  I've found PyInstaller often tries to include other packages I have installed so I make sure my virtualenv is as bare-bones as I can make it.  It also likes to include things like tkinter when I don't need them, so I also use a .spec file for PyInstaller that excludes those specifically.  Here's an example of mine:

```
exclude_list = ['FixTk', 
                'tcl', 
                'tk', 
                '_tkinter', 
                'tkinter', 
                'Tkinter',
                'Tkconstants',
                ]
```

and then in my `Analysis` section I have a line `excludes = exclude_list,`

**Reply by falkoschindler** [reply]
Exactly, 1.3.0 should reduce the package size significantly.

**Reply by CatamountJack** [reply]
With 1.3.2 I was just able to drop 38.6 MB off my PyInstaller packaged program (I still needed pywebview, but I was able to get rid of matplotlib, numpy, and plotly).

**Reply by falkoschindler** [reply]
@CatamountJack So if I understand correctly, you're just shy of 100MB now?

**Comment by CatamountJack** [comment]
Yep - actually for my particular application I'm at 85 MB now.  (I found I was including some other stuff in my old one, so it was really around 125 MB previously.)
